test/unit/org/apache/cassandra/metrics/CounterMutationMetricsTest.java

```diff
@@ -0,0 +1,174 @@
+
+        ColumnFamilyStore cfsOne = Keyspace.open(KEYSPACE1).getColumnFamilyStore(CF1);
+        ColumnFamilyStore cfsTwo = Keyspace.open(KEYSPACE1).getColumnFamilyStore(CF2);
+
+        cfsOne.truncateBlocking();
+        cfsTwo.truncateBlocking();
+
+        Mutation.PartitionUpdateCollector batch = new Mutation.PartitionUpdateCollector(KEYSPACE1, Util.dk("key1"));
+        batch.add(new RowUpdateBuilder(cfsOne.metadata(), 5, "key1")
+                  .clustering("cc")
+                  .add("val", 1L)
+                  .add("val2", -1L)
+                  .build().getPartitionUpdate(cfsOne.metadata()));
+
+        batch.add(new RowUpdateBuilder(cfsTwo.metadata(), 5, "key1")
+                  .clustering("cc")
+                  .add("val", 2L)
+                  .add("val2", -2L)
+                  .build().getPartitionUpdate(cfsTwo.metadata()));
+
+        new CounterMutation(batch.build(), ConsistencyLevel.ONE).apply();
+
+        metrics.assertLockTimeout(0);
+        metrics.assertLocksPerUpdate(1);
+    }
+
+    @Test
+    public void testLockTimeout() throws Throwable
+    {
+        AssertMetrics metrics = new AssertMetrics(11L);
+
+        ColumnFamilyStore cfs = Keyspace.open(KEYSPACE1).getColumnFamilyStore(CF1);
+        cfs.truncateBlocking();
+
+        // Obtain lock
+        new CounterMutation(
+        new RowUpdateBuilder(cfs.metadata(), 5, "key1").clustering("cc")
+                                                       .add("val", 1L)
+                                                       .build(),
+        ConsistencyLevel.ONE).grabCounterLocks(Keyspace.open(KEYSPACE1), new ArrayList<>());
+
+        metrics.assertLockTimeout(0);
+
+        // Try to mutate
+        CompletableFuture.runAsync(() ->
+                                   {
+                                       try
+                                       {
+                                           new CounterMutation(
+                                           new RowUpdateBuilder(cfs.metadata(), 5, "key1")
+                                           .clustering("cc")
+                                           .add("val", 1L)
+                                           .add("val2", -1L)
+                                           .build(),
+                                           ConsistencyLevel.ONE).apply();
+                                       }
+                                       catch (WriteTimeoutException e)
+                                       {
+                                           // Expected exception.
+                                           // If not met, lockTimeout counter will not be increased, which is asserted.
+                                       }
+                                   }).get(100, TimeUnit.MILLISECONDS);
+
+        metrics.assertLockTimeout(1);
+        metrics.assertLocksPerUpdate(2);
+    }
+
+    public class AssertMetrics
+    {
+        public final long timeout;
```

Comment:
`counterWriteRpcTimeoutNanos` ?

Reply:
For the test it's just lock timeout, which is used when `tryLock` is called in `CounterMutation`.